logstash-core/lib/logstash/pipeline.rb

```diff
@@ -156,15 +142,15 @@ def start_workers
       @filters.each {|f| f.register}
 
       pipeline_workers = safe_pipeline_worker_count
-      batch_size = @settings[:pipeline_batch_size]
-      batch_delay = @settings[:pipeline_batch_delay]
+      batch_size = @settings["pipeline.batch.size"]
+      batch_delay = @settings["pipeline.batch.delay"]
       max_inflight = batch_size * pipeline_workers
       @logger.info("Starting pipeline",
-                   :id => self.pipeline_id,
-                   :pipeline_workers => pipeline_workers,
-                   :batch_size => batch_size,
-                   :batch_delay => batch_delay,
-                   :max_inflight => max_inflight)
+                   "id" => self.pipeline_id,
```

Comment:
I don't believe we have a policy.
The reason I chose strings here is the flexibility of them vs symbols, specially since we want to be able to have settings named "pipeline.batch.delay" or "node.name". doing this with symbols would be cumbersome `:"node.name"`

Also, none of this code will be in the critical path performance-wise, so we can afford strings easily.